torch/csrc/jit/passes/quantization.cpp

```diff
@@ -210,12 +207,11 @@ void addQuantDeQuantNodesForInput(
     quant->addInput(qparam_value);
     dequant->addInput(qparam_value);
   }
-  if (t != at::ScalarType::Undefined) {
-    Value* scalartype_v = insertScalarType(quant, t);
-    AT_ASSERT(scalartype_v != nullptr);
-    quant->addInput(scalartype_v);
-    dequant->addInput(scalartype_v);
-  }
+  // Add ScalarType Node
```

Comment:
`addQuantDeQuantNodesForInput` and `addQuantDeQuantNodes` are almost identical. Please consider refactoring them (I suspect you could completely remove one of them if you use `WithInsertPoint`).

Reply:
yes thats planned change now. with zachs change https://github.com/pytorch/pytorch/pull/19788, it allows to insert nodes after param nodes so we do not need special logic anymore

Reply:
done